lucene/test-framework/src/java/org/apache/lucene/tests/store/BaseDirectoryTestCase.java

```diff
@@ -1438,4 +1440,68 @@ public void testListAllIsSorted() throws IOException {
       assertArrayEquals(expected, actual);
     }
   }
+
+  public void testDataTypes() throws IOException {
+    final long[] values = new long[] {43, 12345, 123456, 1234567890};
+    try (Directory dir = getDirectory(createTempDir("testDataTypes"))) {
+      IndexOutput out = dir.createOutput("test", IOContext.DEFAULT);
+      out.writeByte((byte) 43);
+      out.writeShort((short) 12345);
+      out.writeInt(1234567890);
+      out.writeGroupVInts(values, 4);
+      out.writeLong(1234567890123456789L);
+      out.close();
+
+      long[] restored = new long[4];
+      IndexInput in = dir.openInput("test", IOContext.DEFAULT);
+      assertEquals(43, in.readByte());
+      assertEquals(12345, in.readShort());
+      assertEquals(1234567890, in.readInt());
+      in.readGroupVInts(restored, 4);
+      assertArrayEquals(values, restored);
+      assertEquals(1234567890123456789L, in.readLong());
+      in.close();
+    }
+  }
+
+  public void testGroupVInt() throws IOException {
+    try (Directory dir = getDirectory(createTempDir("testGroupVInt"))) {
+      // test fallbackReadGroupVInt
+      doTestGroupVInt(dir, 5, 1, 6, 8);
+
+      // test large data to covers multiple blocks in ByteBuffersDataInput
```

Comment:
adding it explicitly since this is probably the most important case to cover

```suggestion
      // test large data to covers multiple blocks in ByteBuffersDataInput and MMapDirectory via TestMultiMMap
```